### PR TITLE
[Merged by Bors] - TY-2290 feed manager tests

### PR DESCRIPTION
**Pull request by finiteprods**
**Summary**

adds tests to the feed manager.

also moves one method `deactivateDocuments` from the document manager to the feed manager, to remove the dependency between the two managers for easier testing.

### CODE REVIEW

File: discovery_engine/test/feed_manager_test.dart [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      await docBox.clear();
+      await activeBox.clear();
+      await changedBox.clear();
+
+      // reset test data
+      doc2.isActive = true;
+      doc2.feedback = DocumentFeedback.neutral;
+      doc3.isActive = false;
+      doc3.feedback = DocumentFeedback.neutral;
+      data.viewTime.clear();
```
Comment (biernacki) [inline discovery_engine/test/feed_manager_test.dart:110]: you could also avoid this reset by moving objects setup to `setUp` like:

```dart

group('FeedManager', () {
  late Document doc2;
  late Document doc3;

  setUp(() async {
    doc2 = Document(/*...*/);
    doc3 = Document(/*...*/);

    await docRepo.updateMany([doc2, doc3]);
    //...
  });

  tearDown(() async {
    await docBox.clear();
  });

  test('...', {
    // perform test
  });

});
```

**Review by biernacki — APPROVED** [review]

File: discovery_engine/test/feed_manager_test.dart (hunk 1 above)
Comment (finiteprods) [reply discovery_engine/test/feed_manager_test.dart]: nice, thanks :+1:

**Comment by finiteprods** [comment]
bors merge

**Comment by finiteprods** [comment]
bors merge

**Pull request closed without merging**